extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1124,26 +1118,29 @@ export class Bind {
 
   /**
    * Performs CDN rewrites for the given mutation and updates the element.
-   * Returns the rewrite of `value` on success. Otherwise, returns undefined.
    * @see amp-cache-modifications.md#url-rewrites
    * @param {!Element} element
-   * @param {string} property
+   * @param {string} attrName
    * @param {string} value
-   * @return {string|undefined}
+   * @param {boolean} updateProperty If the property with the same name should
+   *    be updated as well.
+   * @return {boolean} Whether or not the rewrite was successful.
    * @private
    */
-  rewriteAttributes_(element, property, value) {
+  rewriteAttributes_(element, attrName, value, updateProperty) {
     // Rewrite attributes if necessary. Not done in worker since it relies on
     // `url#parseUrl` which uses <a>. Worker has URL API but not on IE11.
-    let rewrittenValue;
     try {
-      rewrittenValue = rewriteAttributesForElement(element, property, value);
+      rewriteAttributesForElement(
```

Comment:
I initially though this may need to stay `!!rewriteAttributesForElement`  but I can't find any cases (even cases like Invalid Url) where this method returns falsy (it seems to always return the given initial value if rewriting fails) so this looks fine to me @choumx might know edge cases that I am not aware of. Might be safer to `return !!rewriteAttributesForElement` instead of `true` anyway.

Reply:
Hmm, one question I would have for returning the rewritten value (or cast to boolean) is: what if then updated value is an empty string? In that case, it seems like the associated the property would not be updated as that evaluates to `false`.

I tried the following test case on master and can confirm it fails on the last expect:

```javascript
it('should update properties if the value is an empty string', function* () {
  const {sandbox} = env;
  const spy = sandbox.spy();
  const element = createElement(env, container, '[value]="foo"', 'input');
  yield onBindReadyAndSetState(env, bind, {'foo': 'bar'});
  expect(element.value).to.equal('bar');
  yield onBindReadyAndSetState(env, bind, {'foo': ''});
  expect(element.value).to.equal('');
});
```

I'll add this test case to this change to make sure this doesn't break in the future.

Reply:
We do `String(newVal)` before calling this method for this exact reason.  Fragile... :)

As mentioned, I can't fine any code path where your code could behave differently than previous one based on behaviour of the methods in the call chain. Mabye you do a run as well to double check and then we would be good.

Reply:
never mind `String('')` doesn't make an object and different than `new String('')`. Your change will also fix https://github.com/ampproject/amphtml/issues/14514

Reply:
In this case, I could tell that there was a problem because the type of the return of `rewriteAttributes_` value was `string|undefined`, which means that the value could be the primitive empty string. Can be helpful sometimes to just look at functions and types and reason about those absent the larger context of how things are used.